src/coreclr/System.Private.CoreLib/src/System/StartupHookProvider.CoreCLR.cs

```diff
@@ -12,15 +12,15 @@ namespace System
 {
     internal static partial class StartupHookProvider
     {
-        private static void ManagedStartup()
+        private static unsafe void ManagedStartup(char* pDiagnosticStartupHooks)
         {
 #if FEATURE_PERFTRACING
             if (EventSource.IsSupported)
                 RuntimeEventSource.Initialize();
 #endif
 
             if (IsSupported)
-                ProcessStartupHooks();
+                ProcessStartupHooks(pDiagnosticStartupHooks == null ? string.Empty : new string(pDiagnosticStartupHooks));
```

Comment:
```suggestion
                ProcessStartupHooks(new string(pDiagnosticStartupHooks));
```
`new string` returns `string.Empty` for `(char*)null`